### ISSUE TITLE
Method arguments with leading double-underscore are not getting treated as positional-only

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2497]
## Environment data

-   Language Server version: v2022.3.2
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Sample code

```python
from typing import Any

class C:
    def __eq__(self, other: Any, /) -> bool:
        ...
```

## Expected behavior

No error / warning.

## Actual behavior

![image](https://user-images.githubusercontent.com/15389744/159394845-0ea7bfd0-c97e-417e-a404-75b80137b510.png)

## Additional info

It used to work as expected with Pylance v2022.3.1.

**Comment by Azureblade3808** [comment]
Here is a minimal sample without referencing stubs in typeshed.

```python
class Base:
    def f(self, __a: int) -> None:
        ...

class Derived(Base):
    def f(self, b: int, /) -> None:
        ...
```

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.3.3, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202233-23-March-2022)
